src/ui/component-repository/components/InstallationPlanModal.js

```diff
@@ -0,0 +1,187 @@
+import React, { useState } from 'react';
+import cx from 'classnames';
+import { useSelector, useDispatch } from 'react-redux';
+import { FormattedMessage, injectIntl, intlShape } from 'react-intl';
+import GenericModalContainer from 'ui/common/modal/GenericModalContainer';
+import { Button, Modal, DropdownButton, MenuItem, ButtonGroup } from 'patternfly-react';
+import { Table } from 'react-bootstrap';
+
+import { getInstallPlan, getComponent, getComponentVersion } from 'state/component-repository/components/selectors';
+import { updateInstallPlan, updateAllInstallPlan, installECRComponent, setInstallUninstallProgress } from 'state/component-repository/components/actions';
+import { setVisibleModal } from 'state/modal/actions';
+
+export const MODAL_ID = 'InstallationPlanModal';
+const IGNORED_KEYS = ['version', 'conflictStrategy', 'hasConflicts'];
+const ACTIONS = {
+  SKIP: 'skip',
+  OVERRIDE: 'update',
+  CREATE: 'install',
+};
+
+const InstallationPlanModal = ({
```

Comment:
it's a 150+ LoC component....can we split it?

Reply:
yep, I think I can move the table to its own component, let me see

Reply:
Created a new InstallationPlanTable and split the code accordingly, now this one has around 100 lines.